codegen/method.go

```diff
@@ -1116,35 +1160,39 @@ func (ms *MethodSpec) setWriteQueryParamStatements(
 
 		longQueryName := ms.getLongQueryName(field, thriftPrefix)
 		identifierName := CamelCase(longQueryName) + "Query"
+		_, isList := realType.(*compile.ListSpec)
+		if isList {
+			longQueryName = longQueryName + "[]"
+		}
 
 		if !hasQueryFields {
 			statements.append("queryValues := &url.Values{}")
 			hasQueryFields = true
 		}
 
 		if field.Required {
-			encodeExpr := getQueryEncodeExpression(
-				realType, "r"+longFieldName,
-			)
-
-			statements.appendf("%s := %s",
-				identifierName, encodeExpr,
-			)
-			statements.appendf("queryValues.Set(\"%s\", %s)",
-				longQueryName, identifierName,
-			)
+			if isList {
+				encodeExpr := getQueryEncodeExpression(realType, "value")
+				statements.appendf("for _, value := range %s {", "r"+longFieldName)
+				statements.appendf("\tqueryValues.Add(\"%s\", %s)", longQueryName, encodeExpr)
+				statements.append("}")
+			} else {
+				encodeExpr := getQueryEncodeExpression(realType, "r"+longFieldName)
+				statements.appendf("%s := %s", identifierName, encodeExpr)
+				statements.appendf("queryValues.Set(\"%s\", %s)", longQueryName, identifierName)
+			}
 		} else {
-			encodeExpr := getQueryEncodeExpression(
-				realType, "*r"+longFieldName,
-			)
-
 			statements.appendf("if r%s != nil {", longFieldName)
-			statements.appendf("\t%s := %s",
-				identifierName, encodeExpr,
-			)
-			statements.appendf("\tqueryValues.Set(\"%s\", %s)",
-				longQueryName, identifierName,
-			)
+			if isList {
+				encodeExpr := getQueryEncodeExpression(realType, "value")
```

Comment:
It makes it easier to read what is generated, the code is inside the if clause, so code is not ran twice. Plus it will be up to change once typedef is implemented.